Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextComponent.cpp

```diff
@@ -447,7 +447,7 @@ namespace AzToolsFramework
         else
         {
             loadedSuccessfully = static_cast<PrefabEditorEntityOwnershipService*>(m_entityOwnershipService.get())->LoadFromStream(
-                stream, AZStd::string_view(levelPakFile.toUtf8(), levelPakFile.size()) );
+                stream, AZStd::string_view(levelPakFile.toUtf8().constData(), levelPakFile.size()) );
```

Comment:
This is actually working around an SFINAE issue that GCC is hitting with the `random_access_iterator` concept.

This change is good, but I'll investigate the underlying issue to resolve the root of the problem.

Reply:
The underlying cause is a GCC bug related to examining private functions as part of the evaluating templates: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59002
https://stackoverflow.com/a/48252929

Reply:
From what I read it is clang which is overly permissive here and gcc properly rejects access as access to private methods should not be allowed. Is it correct suggestion?

Reply:
Actually it is the opposite.
GCC is overly permissive and examines private methods when examining members of template parameters during instantiation, when it shouldn't.

The `QByteArray` `operator() QNoImplicitBoolCast() const` function shouldn't be part of the overload set being examined during instantiation, since the template being used does not have access to `QByteArray`